### PR TITLE
feat: cardano preview testnet

### PR DESCRIPTION
**Pull request by vladimirvolek**
Adding new `Cardano Preview Testnet` network.
Legacy testnet will be terminated.

image

image

### CODE REVIEW

**Review by mroz22 — CHANGES_REQUESTED** [review]
If I remember correctly `packages/connect-common/files/coins.json` should be autogenerated from connect-common. So this PR should be accompanied by a corresponding update of submodules

There is some read on this, hope it is up to date
https://github.com/trezor/trezor-suite/blob/develop/docs/packages/connect/supported-coins.md#the-pipeline

File: packages/suite/src/reducers/wallet/cardanoStakingReducer.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const initialState: State = {
         isFetchLoading: false,
         isFetchError: false,
     },
-    testnet: {
+    preview: {
```
Comment (mroz22) [inline packages/suite/src/reducers/wallet/cardanoStakingReducer.ts:28]: can you confirm that we do not need database migration because of these changes?

Comment (vladimirvolek) [reply packages/suite/src/reducers/wallet/cardanoStakingReducer.ts:28]: Yes, it's ok.

**Comment by vladimirvolek** [comment]
> If I remember correctly `packages/connect-common/files/coins.json` should be autogenerated from connect-common. So this PR should be accompanied by a corresponding update of submodules
> 
> There is some read on this, hope it is up to date https://github.com/trezor/trezor-suite/blob/develop/docs/packages/connect/supported-coins.md#the-pipeline

https://github.com/trezor/trezor-common/pull/336

**Comment by vladimirvolek** [comment]
@mroz22 please?

**Comment by mroz22** [comment]
waiting for https://github.com/trezor/trezor-firmware/pull/2716 and submodule update

**Comment by vladimirvolek** [comment]
@mroz22 You are using the old repository as a submodule.

see: https://github.com/trezor/trezor-common/pull/336
https://github.com/trezor/trezor-common
image


```
[submodule "submodules/trezor-common"]
    path = submodules/trezor-common
    url = https://github.com/trezor/trezor-common
```

**Comment by vladimirvolek** [comment]
@mroz22 done

File: packages/connect-common/files/coins.json [hunk 2]
```diff
@@ -110,7 +110,10 @@
             "bech32_prefix": "tb",
             "blockchain_link": {
                 "type": "blockbook",
-                "url": ["https://tbtc1.trezor.io", "https://tbtc2.trezor.io"]
+                "url": [
```
Comment (mroz22) [inline packages/connect-common/files/coins.json]: what produced this formatting changes? is it that we changed our linting rules, or is it some personal setup of yours? this is so hard to read :crying_cat_face:

File: packages/connect-common/files/coins.json [hunk 3]
```diff
@@ -2253,25 +2294,6 @@
             },
             "url": "https://www.rinkeby.io"
         },
-        {
```
Comment (mroz22) [inline packages/connect-common/files/coins.json]: looks like we did not update trezor-common with goerli. 

could you please, @karliatto have a look? also we were about to get rid of tGOR etc. and replace it with tETH instead. can we do it now, or are we still waiting for blockbooks?

**Review by mroz22 — CHANGES_REQUESTED** [review]
ok thanks, 
please keep tGOR record in coins.json - it would break this coin in suite. it looks like we forgot to edit firmware-common when doing this change. 
also I would appreciate not having linting changes unless they are enforced by CI.

File: packages/connect-common/files/coins.json (hunk 3 above)
Comment (sime) [reply packages/connect-common/files/coins.json]: Can we coordinate this for the March release? It's the next release of FW.

Comment (karliatto) [reply packages/connect-common/files/coins.json]: @mroz22

But anyway after some investigations it looks to me that removing of tGOR has already started in trezor-common and that might be a reason why this is removed from here now by the automatic script. On the other side it looks like the version of `submodules/trezor-common` used in this PR is a bit outdated.

So tGOR for goerli was part of trezor-common as you can see here https://github.com/trezor/trezor-common/commit/be351253d36193067b2a4ce81e37b6e623ebed67#diff-478bd47ba2f3d059686166afd3b3630d9d2231e2550454d3e5f1eb64f7f788d0 but it is not anymore.

This is the commit that did the changes https://github.com/trezor/trezor-common/commit/a55191713db27597ee2bd469c2b7e465489e3579#diff-52ccdfbdbb8d7df04c3575c60ece020902c2a4e2f171ce27d826f526ca688c76R39391

But it tGOR is still there in https://github.com/trezor/trezor-common/blob/master/defs/coins_details.json

It looks there are some changes happening in trezor-common, maybe @mmilata can give us more details. And maybe we need to update the `update-coins.sh` script to new version.

Comment (matejcik) [reply packages/connect-common/files/coins.json]: > But it tGOR is still there in https://github.com/trezor/trezor-common/blob/master/defs/coins_details.json

don't look at that file .)

I was under the impression that we decided to _keep_ `tGOR` and friends (bc it makes it easier to see on Trezor on which testnet you are), no?

current master: https://github.com/trezor/trezor-common/blob/master/defs/duplicity_overrides.json#L18-L19

File: packages/connect-common/files/coins.json (hunk 2 above)
Comment (vladimirvolek) [reply packages/connect-common/files/coins.json]: It's from `yarn update-coins` script.

Comment (mroz22) [reply packages/connect-common/files/coins.json]: seems you are right.

**Review by mroz22 — APPROVED** [review]

**Comment by mroz22** [comment]
just please run `yarn format` and push the changes are it is ready for merge

**Comment by vladimirvolek** [comment]
done

File: packages/connect-common/files/coins.json (hunk 3 above)
Comment (sime) [reply packages/connect-common/files/coins.json]: Intention is to sync with upstream and avoid special handling.

**Pull request merged**